symengine/polys/uintpoly.cpp

```diff
@@ -1,7 +1,311 @@
+    dh.itrunc(p);
+    UIntDict ds = UIntDict::from_vec(s.dict_);
+    ds.itrunc(p);
+    UIntDict dt = UIntDict::from_vec(t.dict_);
+    dt.itrunc(p);
+
+    for (unsigned int i = 1; i <= d; ++i) {
+        UIntDict::zz_hensel_step(m, (*this), dg, dh, ds, dt, outArg(dg),
+                                 outArg(dh), outArg(ds), outArg(dt));
+        mp_pow_ui(m, m, 2);
+    }
+
+    res = dg.zz_hensel_lift(p, sub1, l);
+    auto temp = dh.zz_hensel_lift(p, sub2, l);
+    res.insert(res.end(), temp.begin(), temp.end());
+    return res;
+}
+
+void UIntDict::zz_divide(const UIntDict &a, const UIntDict &b,
+                         const Ptr<UIntDict> &quo, const Ptr<UIntDict> &rem)
+{
+    auto b_poly = b;
+    if (a.size() == 0)
+        throw std::runtime_error("Zero Division Error");
+    map_uint_mpz res;
+    UIntDict tmp;
+    integer_class q, r;
+    unsigned int a_deg, b_deg;
+    while (b_poly.degree() >= a.degree()) {
+        a_deg = a.degree();
+        b_deg = b_poly.degree();
+        q = b_poly.get_lc() / a.get_lc();
+        res[b_deg - a_deg] = q;
+        tmp = UIntDict(map_uint_mpz{{b_deg - a_deg, q}});
+        b_poly -= (a * tmp);
+    }
+    *quo = UIntDict(res);
+    *rem = b;
+    *rem -= (a * (*quo));
+}
+
+std::set<RCP<const UIntPoly>, RCPBasicKeyLess> UIntPoly::zz_zassenhaus() const
+{
+    std::set<RCP<const UIntPoly>, RCPBasicKeyLess> factors;
+    unsigned int n = get_degree();
+    UIntDict f(this->poly_);
+    if (n == 1) {
+        factors.insert(rcp_from_this_cast<const UIntPoly>());
+        return factors;
+    }
+    integer_class fc(this->poly_.dict_.begin()->second);
+    auto A = mp_get_ui(this->poly_.max_abs_coef());
+    auto b = get_lc();
+    auto B
+        = std::abs(int(std::sqrt(n + 1)) * std::pow(2, n) * A * mp_get_si(b));
+    auto C = std::pow(n + 1, 2 * n) * std::pow(A, 2 * n - 1);
+    auto gamma = std::ceil(2 * std::log2(C));
+    auto bound = integer_class(int(2 * gamma * std::log(gamma)));
+    std::vector<std::pair<integer_class,
+                          std::set<GaloisFieldDict, GaloisFieldDict::DictLess>>>
+        a;
+    for (integer_class i = 3_z; i <= bound; mp_nextprime(i, i)) {
+        if (b % i == 0)
+            continue;
+        auto F = GaloisFieldDict(this->poly_.dict_, i);
+        if (not F.gf_is_sqf())
+            continue;
+        integer_class temp;
+        F.gf_monic(temp, outArg(F));
+        auto fsqfx = F.gf_zassenhaus();
+        a.push_back({i, fsqfx});
+        if (fsqfx.size() < 15 or a.size() > 4)
+            break;
+    }
+    auto fsqf = *(std::min_element(
+        a.begin(), a.end(),
+        [](const std::pair<integer_class,
+                           std::set<GaloisFieldDict, GaloisFieldDict::DictLess>>
+               &lhs,
+           const std::pair<integer_class,
+                           std::set<GaloisFieldDict, GaloisFieldDict::DictLess>>
+               &rhs) { return lhs.second.size() < rhs.second.size(); }));
```

Comment:
Sorry, didn't read this correctly before. Are you reusing `a` or just using it to get the min element? If so, you can get the min element, without even using the vector.